components/org.wso2.carbon.identity.application.authenticator.iwa/src/main/java/org/wso2/carbon/identity/application/authenticator/iwa/IWAFederatedAuthenticator.java

```diff
@@ -212,31 +212,46 @@ private IWAAuthenticatedUserBean userInformationInListedUserStores(String authen
                                                                        String userStoreDomains)
             throws AuthenticationFailedException {
 
-        String userStoreDomainForUser = null;
         IWAAuthenticatedUserBean authenticatedUserBean = new IWAAuthenticatedUserBean();
-        try {
-            for (String userStoreDomain : userStoreDomains.split(",")) {
-                resolveUserFromUserStore(
-                        authenticatedUserBean, authenticatedUserName, tenantDomain, userStoreDomain.trim());
-                if (authenticatedUserBean.isUserExists()) {
-                    userStoreDomainForUser = userStoreDomain.trim();
-                    break;
+        authenticatedUserBean.setTenantDomain(tenantDomain);
+        authenticatedUserBean.setUser(authenticatedUserName);
+        MultiAttributeLoginService multiAttributeLoginService =
+                IWAServiceDataHolder.getInstance().getMultiAttributeLoginService();
+
+        if (multiAttributeLoginService.isEnabled(tenantDomain)) {
+            ResolvedUserResult resolvedUserResult = multiAttributeLoginService.resolveUser(
+                    authenticatedUserName, tenantDomain);
+            if (resolvedUserResult != null &&
```

Comment:
If there is no user found, authenticatedUserBean will be returned as userExists = false ? and the later logic will handle there is no result found via userExists variable ?

Reply:
Yes, this is handled here https://github.com/wso2-extensions/identity-local-auth-iwa-kerberos/blob/48bf34fd07cb6748cba41a8efa42d856aee38c82/components/org.wso2.carbon.identity.application.authenticator.iwa/src/main/java/org/wso2/carbon/identity/application/authenticator/iwa/IWAFederatedAuthenticator.java#L123